src/subprocess-posix.cc

```diff
@@ -29,6 +29,19 @@ extern char** environ;
 
 #include "util.h"
 
+int HandleEAGAINPosixSpawn(pid_t* pid, const char* path,
+                           const posix_spawn_file_actions_t* file_actions,
+                           const posix_spawnattr_t* attrp, char* const argv[],
+                           char* const envp[]) {
+  while (true) {
+    int ret = posix_spawn(pid, path, file_actions, attrp, argv, envp);
+    if (ret == -1 && errno == EAGAIN)
+      sleep(1);
```

Comment:
I'm not sure silently being slow is better than failing quickly. This happens if you oversubscribe your build box (like it happened on that cros bug where this is from); a better fix to me seems to fix the bot instead.

I'd kind of buy the patch without the sleep, for spec compat reasons, but imho if you run into a situation where you need this, something already went wrong earlier.

Reply:
At the very least some kind of logging (warning) should happen.

Reply:
I'd point out that running the loop indefinitely is also not a great choice. At worst, put a limit of 1000 retries or something, so it doesn't just continue failing forever.

Reply:
I think the right answer here is not obvious. Clearly everyone has a good point. However, unless someone is intimately familiar with the semantics of posix_spawn(), they will expect it to block until it is complete. Performance vs. Correctness is another question. Again, this comes down to "what is the user expectation when they run ninja?" Do the expect ninja to be mindful of their system resources and not retry upon any sort of system-level hiccup (thus pushing the design in Linux up to the user) or do they just want the build to push through, waiting until it has a chance to run?

How would folks feel if I plumbed a ninja command-line option (and/or env var) down here so that users could elect one behavior or another?

Reply:
I think that for purposes of this PR, it might be better for you to make the policy be a parameter to the Subprocess::Start() function, and default it to the current behavior.

Then once the Ninja project figures out how it wants to handle environment variables, and/or commandline options, it can (potentially) be added later.

I don't think that adding a new cli parameter or environment variable will be accepted at this time.

Reply:
Consider also consulting the behavior that other implementations of "Spawn a process" use. E.g. Boost.Process (https://github.com/boostorg/process/blob/develop/include/boost/process/detail/posix/executor.hpp)

I'm not saying that I have an answer for you, simply that consulting other implementations may give Ninja some wisdom on potential pitfalls.

Reply:
I would expect the build to push through, waiting until it has a chance to run. IMHO your patch is fine, it should just not do this silently, but warn the user.